pydantic/datetime_parse.py

```diff
@@ -23,7 +23,9 @@
 date_re = re.compile(r'(?P<year>\d{4})-(?P<month>\d{1,2})-(?P<day>\d{1,2})$')
 
 time_re = re.compile(
-    r'(?P<hour>\d{1,2}):(?P<minute>\d{1,2})(?::(?P<second>\d{1,2})(?:\.(?P<microsecond>\d{1,6})\d{0,6})?)?'
+    r'(?P<hour>\d{1,2}):(?P<minute>\d{1,2})'
+    r'(?::(?P<second>\d{1,2})(?:\.(?P<microsecond>\d{1,6})\d{0,6})?)?'
+    r'(?P<tzinfo>Z|[+-]\d{2}(?::?\d{2})?)?$'
```

Comment:
```py
tz_suffix = r'(?P<tzinfo>Z|[+-]\d{2}(?::?\d{2})?)?$'

time_re = re.compile(... + tz_suffix)
...
datetime_re = re.compile(... + tz_suffix)
```

and reuse the reusable bit

Reply:
in fact you might be able to reuse the whole of the time regex at the end of `datetime_re`?